Cn0415 demo dev

adrimbarean opened the pull request:
Initial commit for the CN0415 project: Solenoind Closed Loop Control Demo

https://wiki.analog.com/resources/eval/user-guides/eval-adicup3029/reference_designs/demo_cn0415

adrimbarean commented:
V0 -> V1:
- fix travis build errors
- fix last commit to only show the PID module and not the AStyle patch

stefpopa commented:
As a general remark, the title of your commits should describe your changes in an imperative mood, e.g.
`cn0415: Implement PID controller module`

adrimbarean commented:
V1 -> V2:
 -  reorder commits to bottom-top approach
 - add name of the ADC part (LTC2312-14) in the commit title
 - make titles more imperative!

mirceacaprioru commented on projects/ADuCM3029_demo_cn0415/src/platform_drivers.c:
```diff
@@ -0,0 +1,741 @@
+	ADI_SPI_TRANSCEIVER m_spi_transceive;
+	ADI_SPI_RESULT e_spi_result;
+
+	m_spi_transceive.pTransmitter = NULL;
```
the name of the functions suggests that this is also used to write to spi. data buffer could also be placed on the transmitter and after the functions data will contain the read information.

mirceacaprioru commented on projects/ADuCM3029_demo_cn0415/src/platform_drivers.c:
```diff
@@ -0,0 +1,741 @@
+	if (!dev)
+		return FAILURE;
+
+	dev->type	= ADICUP3029_GPIO;
```
nitpick: space after variable name not tab.

mirceacaprioru commented on projects/ADuCM3029_demo_cn0415/src/platform_drivers.c:
```diff
@@ -0,0 +1,741 @@
+	const uint8_t baudrate_divm_26mhz[] = {3, 3, 2, 3, 1, 1, 1, 1, 1, 1};
+	const uint16_t baudrate_divn_26mhz[] = {1078, 1078, 1321, 1078, 1563, 1563,
+						1563, 1563, 1280, 171
+					       };
```
nitpick: this could be placed on the previous line.

adrimbarean replied:
The AStyle places it this way.

stefpopa replied:
AStyle is not the king of the world :). We should take a second look at it's fixes.

mirceacaprioru commented on projects/ADuCM3029_demo_cn0415/RTE/Device/ADuCM3029/adi_wdt_config.h:
```diff
@@ -0,0 +1,119 @@
+ *  @ingroup WDT_Driver
+ *  @{
+ */
+
```
nitpick: extra line

stefpopa commented on projects/ADuCM3029_demo_cn0415/src/platform_drivers.c:
```diff
@@ -0,0 +1,741 @@
+	port = (desc->number & 0xF0) >> 4;
+
+	/* Change pin value */
+	if(value == 1)
```
why not check GPIO_HIGH ?

stefpopa commented on projects/ADuCM3029_demo_cn0415/src/platform_drivers.c:
```diff
@@ -0,0 +1,741 @@
+	/* Values for the baudrate generation registers when the baudrate is one of
+	 * the standard values and clock is 26MHz */
+	const uint8_t baudrate_osr_26mhz[] = {3, 3, 3, 3, 3, 3, 3, 2, 2, 2};
+	const uint16_t baudrate_divc_26mhz[] = {24, 12, 8, 4, 4, 2, 1, 1, 1, 1};
```
it seems that uint8_t should be enough

stefpopa commented on projects/ADuCM3029_demo_cn0415/src/cn0415.c:
```diff
@@ -0,0 +1,2189 @@
+
+	/* Get character and identify it */
+	ret = cn0415_parse(dev);
+	if(ret !
```
this is pretty difficult to read. Maybe rename the local variables to make them shorter: `ext_measurement_current -> ext_meas_cur` or something

stefpopa commented on projects/ADuCM3029_demo_cn0415/src/cn0415.c:
```diff
@@ -0,0 +1,2189 @@
+
+	/* Save frequency in flash */
+	dev->param_array[F_PWM] = ifreq;
+	return flash_write(dev->param_array, FLASH_PARAM_SIZE);
```
nitpick: I think the convention is to leave a white line before return:
```
	dev->param_array[F_PWM] = ifreq;

	return flash_write(dev->param_array, FLASH_PARAM_SIZE);
```

Check this everywhere in your code

stefpopa commented on projects/ADuCM3029_demo_cn0415/src/cn0415.c:
```diff
@@ -0,0 +1,2189 @@
+	ret = timer_start();
+	if(ret != 0)
+		goto error;
+#ifdef FLASH_INIT_PARAMETERS
```
Curious: do we really have to use conditional groups?

adrimbarean replied:
I only did it this way to follow the example projects in the no-OS repo. If there is a better way to do it I want to do that.

adrimbarean replied:
We need the program to have ways to initialize the application: from the main function in the ADuCM_demo_cn0415.c or from the flash. Also to have a easy way to change the flash initialization values.

adrimbarean commented:
V2 -> V3:
 - updated .astyleignore file;
 - updated spi_write_and_read function write too, not just read;
 - corrected end bracket bracket new line on initialized arrays (showed AStyle who's boss);
 - replaced tabs with spaces in appropriate places;
 - check for define instead of magic number on GPIO functions;
 - ensure empty new lines before returns;
 - made local variables shorter and added comment explanation;
 - changed uint16_t to uint8_t on uart setup baudrate generation arrays.

adrimbarean commented:
V3 -> V4:
 - LCT2312 ADC driver was successfully merged into no-OS repo

adrimbarean commented:
Will merge today if everything is good to go.
